test/models/concerns/organization_extensions_test.rb

```diff
@@ -26,5 +26,24 @@ def test_audit_manifest_action
       end
       assert_equal @org.manifest_refreshed_at.to_i, current_time.to_i
     end
+
+    def test_clear_syspurpose_status
+      host = FactoryBot.create(:host)
+
+      HostStatus::Status.create(id: 1, host_id: host.id, type: 'Katello::PurposeStatus')
+      HostStatus::Status.create(id: 2, host_id: host.id, type: 'Katello::PurposeAddonsStatus')
+      HostStatus::Status.create(id: 3, host_id: host.id, type: 'Katello::PurposeUsageStatus')
+      HostStatus::Status.create(id: 4, host_id: host.id, type: 'Katello::PurposeRoleStatus')
+      HostStatus::Status.create(id: 5, host_id: host.id, type: 'Katello::PurposeSlaStatus')
+
+      assert_equal HostStatus::Status.count, 5
+
+      Organization.any_instance.stubs(:simple_content_access?).returns(true)
+
+      host_purpose = HostStatus::Status.where(type: ['Katello::PurposeStatus','Katello::PurposeAddonsStatus','Katello::PurposeUsageStatus','Katello::PurposeRoleStatus','Katello::PurposeSlaStatus'])
```

Comment:
I think you meant to test your `clear_syspurpose_status` method? ie `some_organization.clear_syspurpose_status`

Looks like this test class already has an org to use defined in its setup method!

Reply:
Fixed and updated, based on what we talked about. I will start the refactor work that you mentioned above shortly.